### PR TITLE
Imbalance and run bars proskurin alex

### PR DESCRIPTION
**Pull request by proskurin**
Implement imbalance and run bars (dollar, volume, tick) implementation. Added fast_ewma module for ultrafast ewma implementation as a result numba (0.43.0) dependency was added. Created unit -tests for imbalance_data_structures.py and run_data_data_structures and fast_ewma.py, updated requirements.txt (added numba dependency).
Reimplement all data structures with namedtuples in cache instead of arrays

Fixes # (issue)
Problem of high/low price calculation in data_structures.py (change 'elif' to 'if' in min/max check)

### CODE REVIEW

File: mlfinlab/data_structures/data_structures.py [hunk 1]
```diff
@@ -30,11 +31,11 @@ def _update_counters(cache, flag):
     # Check flag
     if flag and cache:
         # Update variables based on cache
-        cum_ticks = int(cache[-1][6])
-        cum_dollar_value = np.float(cache[-1][5])
-        cum_volume = cache[-1][4]
-        low_price = np.float(cache[-1][2])
-        high_price = np.float(cache[-1][3])
+        cum_ticks = int(cache[-1].cum_ticks)
```
Comment (roaffix) [inline mlfinlab/data_structures/data_structures.py]: `cache[-1]` is not intuituve. Consider to write it into separate variable and leave short inline comment

File: mlfinlab/data_structures/data_structures.py [hunk 2]
```diff
@@ -106,13 +117,16 @@ def _assert_dataframe(test_batch):
     :param test_batch: DataFrame which will be tested.
     """
     assert test_batch.shape[1] == 3, 'Must have only 3 columns in csv: date_time, price, & volume.'
-    assert isinstance(test_batch.iloc[0, 1], float), 'price column in csv not float.'
-    assert isinstance(test_batch.iloc[0, 2], np.int64), 'volume column in csv not int.'
+    assert isinstance(test_batch.iloc[0, 1],
```
Comment (roaffix) [inline mlfinlab/data_structures/data_structures.py]: IMO, it's a bad idea to use indexes. A little change can lead to huge bugs. Is it possible to change index references to name references ?

File: mlfinlab/tests/test_fast_ewma.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+class TestDataStructures(unittest.TestCase):
+    """
+    Test the various financial data structures:
+    1. Imbalance Dollar bars
+    2. Imbalance Volume bars
+    3. Imbalance Tick bars
+    """
+
+    def setUp(self):
```
Comment (roaffix) [inline mlfinlab/tests/test_fast_ewma.py]: In python you shouldn't name functions like this `thisIsMyFunctions`. Use underscores to separate the words
P.S. in this case it'd be better to name function `set_tick_data_path` or something. Don't be scared to use long names. You read code more often than write it :)

Comment (roaffix) [reply mlfinlab/tests/test_fast_ewma.py]: FYO, check the library `pathlib`

Comment (Jackal08) [reply mlfinlab/tests/test_fast_ewma.py]: Hi @roaffix, 

Totally agree. Snake case is the way to do it, however in this case the method SetUp() is a special method used in the python unit testing framework to signal to the unit tests which functionality should be run before each unit test.

Comment (roaffix) [reply mlfinlab/tests/test_fast_ewma.py]: MB. Missed that. Thanks for a reply :)

**Pull request closed without merging**